### DISCUSSION TITLE
Commit made by a non-existent account

### DISCUSSION THREAD

**Discussion by Argald0** [community/community#23155] [New to GitHub]
<h1>My commit’s author is a non-existent account instead of mine.</h1>
<p>I don’t really know where I’m supposed to write that, or if it will help anyone, but I feel like it should be written somewhere.</p>
<h3>Deleted old account and made a new one</h3>
<p>The title says it all. During my studies I made a Github account with my school mail address, that I didn’t want to keep anymore, so I decided to <strong>delete that old account</strong>, and make a brand new with my own personal mail. <em>While I had the old account, I installed and used Git for Windows.</em></p>
<p>I deleted properly my previous account, by going to the settings of this account, asking for it to be deleted, writing the little sentence acknowledging that I really want to delete it, and a message appears saying that : “<em>The account was officially deleted, and all information concerning this account will be deleted in the following 24hrs.</em>” (or something like this)</p>
<h3>The problem</h3>
<p>So I made a new account a few days ago, and thought I’d start exploring Github, but also use it to its full potential. I decided to make a MkDocs site based on Github Pages, and created a repo for that purpose. Then, I imported the repo using VS Code, and started modifying it, saving changes, committing, pushing and so on, in order to get used to it, as I never used it before. When I went on Github to check the changes, the commits were made by… My Old Account :interrobang: As I deleted the previous account, I was wondering how it was possible that it was committed by that same account !</p>
<p>I searched on the forum, and found out that there are 3 separate roles (the author, the committer and the pusher) thanks to this great topic : <a href="https://github.community/t/why-is-my-commit-associated-with-the-wrong-person/10199" class="inline-onebox">Why is my commit associated with the wrong person?</a><br>
As I checked who was the author of my commit using the following command : <code>git show &lt;SHA&gt;</code> it displayed my previous account (the deleted one) information : nickname and mail address. This is where I realised that the error in the nickname comes directly from me, from the changes on my PC.<br>
Searching a bit more, I found out that when committing and pushing, Git uses the information that are stored in the global file <code>C:\Users\&lt;username&gt;\.gitconfig</code>, which still had my previous nickname in it, and the associated mail address. All good, I changed it, and now my commits are from the right account ! :grinning_face_with_smiling_eyes:</p>
<h3>What’s important</h3>
<p>Doing my researches, I found out that the nickname in this file is a bit useless as Github will resolve itself the nickname associated to the mail address noted in the file. This is where my questioning stands : how is it possible for Github to match this supposedly dead account to this mail address it the account is deleted, and if all data concerning the account are deleted (implying even the mail address) ?</p>
<h4>Finaly</h4>
<p>Thanks for reading ! :wink:</p>

**Comment by canuckjacq** [comment, marked as answer]
<aside class="quote no-group" data-username="Argald0" data-post="1" data-topic="161137">
<div class="title">
<div class="quote-controls"></div>
 Argald0:</div>
<blockquote>
<p>I found out that the nickname in this file is a bit useless as Github will resolve itself the nickname associated to the mail address noted in the file.</p>
</blockquote>
</aside>
<blockquote>
<p>how is it possible for Github to match this supposedly dead account to this mail address it the account is deleted, and if all data concerning the account are deleted</p>
</blockquote>
<p>If we can’t match the commit email address to an account, the nickname on the commit is used instead. So it’s not matching your old email address with your deleted account - it’s using the nickname you used in those commits.</p>

**Comment by Argald0** [comment]
<p>Makes sense ! :stuck_out_tongue:<br>
Thank you for your explanation, seems much clearer now ! :wink:</p>
